flycheck.el

```diff
@@ -6330,6 +6331,118 @@ See URL `http://lesscss.org'."
+  :safe (lambda (value) (member value flycheck-luacheck-std-globals-list))
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-globals nil luacheck
+  "Filter out warnings related to global variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused nil luacheck
+  "Filter out warnings related to unused variables and values."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-redefined nil luacheck
+  "Filter out warnings related to redefined variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused-args nil luacheck
+  "Filter out warnings related to unused arguments and loop variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused-secondaries nil luacheck
+  "Filter out warnings related to unused variables set together
+with used ones."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused-globals nil luacheck
+  "Filter out warnings related to set but unused global variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-globals nil luacheck
+  "A list of additional globals to add on top of standard ones."
+  :type '(choice (const :tag "No additional globals" nil)
+                 (repeat :tag "Additional global"
+                         (string :tag "Global name")))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-read-globals nil luacheck
+  "A list of additional globals to add on top of standard ones."
+  :type '(choice (const :tag "No additional read-only globals" nil)
+                 (repeat :tag "Additional read-only global"
+                         (string :tag "Global name")))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-define-checker luacheck
+  "A Lua syntax checker using luacheck.
+
+See URL `https://github.com/mpeterv/luacheck'."
+  :command ("luacheck"
+            "--no-config"
+            (option "--std" flycheck-luacheck-std-globals)
+            (option-flag "--no-global" flycheck-luacheck-no-globals)
+            (option-flag "--no-unused" flycheck-luacheck-no-unused)
+            (option-flag "--no-redefined" flycheck-luacheck-no-redefined)
+            (option-flag "--no-unused-args" flycheck-luacheck-no-unused-args)
+            (option-flag "--no-unused-secondaries" flycheck-luacheck-no-unused-secondaries)
+            (option-flag "--no-unused-globals" flycheck-luacheck-no-unused-globals)
+            "--formatter" "plain"
+            "--codes"
```

Comment:
@ptrv Would you mind to add a small comment that explains the use of `--codes` here?  This would make it easier for me to understand what's going on later.